src/Expr.cc

```diff
@@ -3199,7 +3199,26 @@ void IndexExpr::Assign(Frame* f, Val* v, Opcode op)
 
 	switch ( v1->Type()->Tag() ) {
 	case TYPE_VECTOR:
-		if ( ! v1->AsVectorVal()->Assign(v2, v, op) )
+		{
+		const ListVal *lv = v2->AsListVal();
+
+		if ( lv->Length() > 1 )
+			{
+			int len = v1->AsVectorVal()->Size();
+			bro_int_t first = get_slice_index(lv->Index(0)->CoerceToInt(), len);
+			bro_int_t last = get_slice_index(lv->Index(1)->CoerceToInt(), len);
+			int slice_length = last - first;
+
+			const VectorVal *v_vect = v->AsVectorVal();
+			if ( slice_length != v_vect->Size())
+				RuntimeError("vector being assigned to slice does not match size of slice");
```

Comment:
Python is happy to let you assign a vector to a slice that doesn't have the same size as the slice. For example,
```python
>>> a = [1, 2, 3]
>>> a[0:2] = a
>>> a
[1, 2, 3, 3]
```
is strange, but works fine. I think making this work in Zeek would require that we rebuild the vector because you'd have to resize it, but might be worthwhile. I could see a use case where someone would like to insert a vector into another one and want to use this syntax. What do you think about that?

Reply:
I think I need to refresh my knowledge of slicing in Python, as you're finding all sorts of side cases I didn't think about 😄 . It also works the other direction where you can shrink a vector with slicing:

```
>>> a = [1, 2, 3]
>>> a[0:2] = [4]
>>> a
[4, 3]
```

Reply:
For the ultimate challenge we could try and make something like this work :--)
```python
>>> a = [1, 2, 3]
>>> a.append(a)
>>> a[0:1] = a
>>> a
[1, 2, 3, [...], 2, 3, [...]]
```
What you have now looks really good though. I think its mostly just a matter of preference if we actually allow for weird resizing behaviour.

Reply:
> For the ultimate challenge we could try and make something like this work :--)

Our vectors are typed and all elements must be of homogenous type.